javascript/javascript1/week2/index.js

```diff
@@ -0,0 +1,234 @@
+  }
+  if (sirName.length > 0) {
+    //sirName = sirName.trim();
+    sirName = sirName.replace(/\s/g, "");
+  }
+  if (firstName === "" && sirName === "") {
+    //throw new console.error("First name and sir name should not be empty");
+    console.warn("First name and sir name should not be empty");
+    //throw "First name and sir name should not be empty";
+    return;
+  }
+  if (!(useFormalName === true || useFormalName === false)) {
+    //throw "Invalid parameter, bolean expected";
+    console.warn("Invalid parameter, bolean expected");
+    return;
+  }
+
+  if (useFormalName) {
+    if (gender === Male) {
+      return `Lord ${firstName} ${sirName}`;
+    } else if (gender === Female) {
+      return `Lady ${firstName} ${sirName}`;
+    }
+  }
+  return `${firstName} ${sirName}`;
+}
+
+const fullname1 = getFullname("Reenu", "Abraham");
+if (fullname1 != undefined) {
+  console.log(fullname1);
+}
+const fullname2 = getFullname("Benjamin", "Hughes", 5);
+console.log(fullname2);
+const fullname3 = getFullname("Sofia", "John", true, "abcd");
+console.log(fullname3);
+const fullname4 = getFullname("Catrine", "", true, Female);
+if (fullname4 != undefined) {
+  console.log(fullname4);
+}
+
+const fullname5 = getFullname(" ", "", true, Female);
+if (fullname4 != undefined) {
+  console.log(fullname4);
+}
+
+//Ex 2
+const weekDay = {
+  0: "Sunday",
+  1: "Monday",
+  2: "Tuesday",
+  3: "Wednesday",
+  4: "Thursday",
+  5: "Friday",
+  6: "Saturday",
+};
+
+function getEventWeekday(noOfdaysFromToday) {
+  const day = new Date();
+  day.setDate(day.getDate() + noOfdaysFromToday);
+  return weekDay[day.getDay()];
+}
+
+console.log(`Event day is ${getEventWeekday(40)}`);
+
+//Ex3
+
+function getClothingRecommendation(temperatureInCelcius) {
+  if (temperatureInCelcius < 0) {
+    return "4 layers of winter clothes";
+  }
+  if (temperatureInCelcius <= 10) {
+    return "3 layers of winter clothes";
+  }
+  if (temperatureInCelcius > 10 && temperatureInCelcius <= 15) {
+    return "Spring wear";
+  }
+  if (temperatureInCelcius > 15 && temperatureInCelcius <= 20) {
+    return "Summer wear with & a small coat";
+  }
+  if (temperatureInCelcius > 20) {
+    return "summer wear";
+  }
+}
+const clothesToWear = getClothingRecommendation(18);
+console.log(clothesToWear);
+
+// Ex 4
+const class07Students = [];
+function addStudentToClass(studentName) {
+  studentName = studentName.replace(/\s/g, "");
+  if (studentName.length <= 0) {
+    console.log("Please enter a student name");
+    return;
+  }
+  if (!isStudentInclass(studentName)) {
+    if (getNumberOfStudents() < 6 || studentName.toLowerCase() === "queen") {
+      class07Students.push(studentName);
+    } else {
+      console.log("Cannot accept more students as the class is already full");
+    }
+  } else {
+    console.log(`${studentName} is already added to the class`);
+  }
+}
+
+function isStudentInclass(studentName) {
+  return class07Students.find(
```

Comment:
'includes' is case sensitive. so I can repeat names "Reenu" , "reenu".  otherwise I have to loop through the array. so isn't better with find?

Reply:
You can use toLowerCase() as you are using here. You can write 
`return class07Students.find(name => name.toLowerCase() === studentName.toLowerCase());`
Your choice of course, I just wanted to suggest using includes, since it is usually used if you need to find if a value exists in an array.

Reply:
Thank you so much Svetlana for reviewing the code and taking time time to reply my doubts. Thank you for suggesting 'includes' ..I tried that its working for lowercases as well .. will use that....